packages/core/src/physics/PhysicsManager.ts

```diff
@@ -232,7 +264,21 @@ export class PhysicsManager {
    * @internal
    */
   _update(deltaTime: number): void {
-    this._nativePhysicsManager.update(deltaTime);
+    const {
+      _fixedTimeStep: fixedTimeStep,
+      _maxStepCount: maxStepCount,
+      _restTime: restTime,
+      _nativePhysicsManager: nativePhysicsManager
+    } = this;
+    const componentManager = this._engine._componentsManager;
+
+    const simulateTime = deltaTime + restTime;
+    const step = Math.min(maxStepCount, Math.floor(simulateTime / fixedTimeStep));
+    this._restTime = simulateTime - step * fixedTimeStep;
+    for (let i = 0; i < step; i++) {
+      componentManager.callScriptOnPhysicsUpdate();
+      nativePhysicsManager.update(fixedTimeStep);
+    }
   }
```

Comment:
Simplified code:
```ts
_update(deltaTime: number): void {
    const { _fixedTimeStep: fixedTimeStep, _nativePhysicsManager: nativePhysicsManager } = this;
    const { _componentsManager: componentManager } = this._engine;

    const simulateTime = deltaTime + this._restTime;
    const step = Math.min(this._maxStepCount, Math.floor(simulateTime / fixedTimeStep));
    this._restTime = simulateTime - step * fixedTimeStep;
    for (let i = 0; i < step; i++) {
      componentManager.callScriptOnPhysicsUpdate();
      nativePhysicsManager.update(fixedTimeStep);
    }
  }
```

Reply:
+1

Reply:
1.Why not use milliseconds and just pass in seconds when calling nativePhysicsManager.update？
2.Why
```
      componentManager.callScriptOnPhysicsUpdate();
      nativePhysicsManager.update(fixedTimeStep);
```
instead of
```
      nativePhysicsManager.update(fixedTimeStep);
      componentManager.callScriptOnPhysicsUpdate();
```

Reply:
**Mark**: maybe need lateUpdate